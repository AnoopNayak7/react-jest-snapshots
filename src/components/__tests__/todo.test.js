import {render, screen, cleanup} from '@testing-library/react'
import Todo from '../todo';
import renderer from 'react-test-renderer'

afterEach(() => {
    cleanup()
})
test('should render todo component that is not completed', () => {
    const todo =  { id: 1, title: "wash dishes", completed: false }
    render(<Todo todo={todo} />);
    const tododElement = screen.getByTestId('todo-1');
    expect(tododElement).toBeInTheDocument();
    expect(tododElement).toHaveTextContent("wash dishes")
    expect(tododElement).not.toContainHTML('<strike>')
})

test('should render todo component that is completed', () => {
    const todo =  { id: 2, title: "wash car", completed: true }
    render(<Todo todo={todo} />);
    const tododElement = screen.getByTestId('todo-2');
    expect(tododElement).toBeInTheDocument();
    expect(tododElement).toHaveTextContent("wash car");
    expect(tododElement).not.toContainHTML('<strike>')
});

test('match snapshots', () => {
    const todo =  { id: 2, title: "wash car", completed: false };
    const tree = renderer.create(<Todo todo={todo} />).toJSON();
    expect(tree).toMatchSnapshot();
})