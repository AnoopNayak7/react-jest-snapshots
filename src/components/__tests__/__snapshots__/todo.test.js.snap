// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshots 1`] = `
<div
  data-testid="todo-2"
>
  <div>
    <h1>
      wash car
    </h1>
  </div>
</div>
`;
